File: Day33_RemDup2.cpp
#include <bits/stdc++->h>
using namespace std;
ifstream in("in->txt");
ofstream out("out->txt");

ListNode *deleteDuplicates(ListNode *head)
{
    ListNode *newn = ListNode(0, nullptr);
    auto prev = newn;
    auto temp = head;
    while (temp)
    {
        if (temp->next == nullptr or temp->val != temp->next->val)
        {
            prev->next = temp;
            prev = temp;
        }
        else
            while (temp and temp->next and temp->val == temp->next->val)
                temp = temp->next;
        temp = temp->next;
    }
    prev->next = nullptr;
    return newn->next;
}

int main()
{

    return 0;
}

File: Day33_RemDup2.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        newn = ListNode(0, None)
        prev = newn
        temp = head
        while(temp):
            if(temp.next == None or temp.val != temp.next.val):
                prev.next = temp
                prev = temp
            else:
                while(temp and temp.next and temp.val == temp.next.val):
                    temp = temp.next
            temp = temp.next
        prev.next = None
        return newn.next